extensions/wikia/Flags/Flags.hooks.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Flags;
+
+class Hooks {
+
+	/**
+	 * Hooks into the internalParse() process and injects a wikitext
+	 * with notices for the given page.
+	 * @param \Parser $parser
+	 * @param $text
+	 * @param $stripState
+	 * @return bool
+	 */
+	public static function onParserBeforeInternalParse( \Parser $parser, &$text, &$stripState ) {
+		global $wgRequest;
+
+		/**
+		 * Don't check for flags if:
+		 * - you've already checked
+		 * - a user is on an edit page
+		 * - the request is from VE
+		 */
+		if ( !$parser->mFlagsParsed
+			&& !\WikiaPageType::isEditPage()
+			&& !( $wgRequest->getVal( 'action' ) == 'visualeditor' )
```

Comment:
The "internal" API calls from VE don't have the `veaction` parameter but `action=visualeditor` and `paction=parse`.